src/main/java/com/laserfiche/api/client/APIServer/TokenClientImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.laserfiche.api.client.APIServer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.laserfiche.api.client.model.ApiException;
+import com.laserfiche.api.client.model.CreateConnectionRequest;
+import com.laserfiche.api.client.model.OffsetDateTimeDeserializer;
+import com.laserfiche.api.client.model.ProblemDetails;
+import com.laserfiche.api.client.oauth.TokenClientObjectMapper;
+import kong.unirest.Header;
+import kong.unirest.HttpResponse;
+import kong.unirest.Unirest;
+import kong.unirest.json.JSONObject;
+import org.threeten.bp.OffsetDateTime;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class TokenClientImpl implements TokenClient {
+    private String baseUrl;
+
+    protected ObjectMapper objectMapper;
+
+    public TokenClientImpl(String baseUrl) {
+        this.baseUrl = baseUrl;
+        Unirest
+                .config()
+                .setObjectMapper(new TokenClientObjectMapper());
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(OffsetDateTime.class, new OffsetDateTimeDeserializer());
+        this.objectMapper = JsonMapper
+                .builder()
+                .addModule(module)
+                .disable(MapperFeature.CAN_OVERRIDE_ACCESS_MODIFIERS)
+                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+                .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+                .build();
+    }
+
+    @Override
+    public CompletableFuture<SessionKeyInfo> createAccessToken(String repoId, CreateConnectionRequest body) {
+        Map<String, Object> pathParameters = getNonNullParameters(new String[]{"repoId"},
+                new Object[]{repoId});
+        CompletableFuture<HttpResponse<SessionKeyInfo>> future = Unirest
+                .post(baseUrl + "/v1/Repositories/{repoId}/Token")
+                .routeParam(pathParameters)
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .field("grant_type", "password")
+                .field("username", body
+                        .getUsername()
+                        .toString())
+                .field("password", body
+                        .getPassword()
+                        .toString())
+                .asObjectAsync(SessionKeyInfo.class);
```

Comment:
Would this work? If we deserialize it into `SessionKeyInfo` but it's actualluy a JSON corresponds to `ProblemDetails`, I think it would throw exception like not being able to fit the JSON into `SessionKeyInfo`.

Reply:
ok so i just tried to match the latest version of the auto generated code in the java repo client lib